### PR TITLE
Markdown linter

### PR DESCRIPTION
**Pull request by mrsimpson**
## 📚 Purpose
Adds a markdown linter

## 👌 Resolves:
The need for manual markdown analysis

## 📦 Impacts:
In order to keep the impact small, I limited the linter's scope to the top-level Readme. I believe it should be enabled for all own markdown-files ( `markdownlint '**/*.md' --ignore node_modules` ), but this would have quite some impact which I as a new contributor didn't want to throw at you ;)

## ✅ PR Checklist
- [X] Simple markdown-configuration for the main Readme
- [x] Decide about the scope (-> All non-generated MD-files)
- [x] Decide about [pre-commit](https://www.npmjs.com/package/markdownlint-cli#user-content-use-with-pre-commit) (-> Part of the default linting target => wherever there is lint, there shall be markdown linting too ;) )
- [x] Integrate into CI (-> 🔝 )

### CODE REVIEW

**Review by avatarneil — APPROVED** [review]

**Review by drtyh2o — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "lint": "pnpm lint:audit && pnpm lint:prettier '**/package.json' && pnpm lint:eslint '**/*.ts'",
```
Comment (PeterSP) [inline package.json:20]: Do we want to add `lint:markdown` to `lint` here?
```suggestion
    "lint": "pnpm lint:audit && pnpm lint:prettier '**/package.json' && pnpm lint:eslint '**/*.ts' && pnpm lint:markdown",
```

**Comment by mrsimpson** [comment]
@PeterSP what do you think about the scope of linting md-files?

**Comment by PeterSP** [comment]
> @PeterSP what do you think about the scope of linting md-files?

Do you mean whether it should be applied to other markdown files besides the root readme? We have a number of CHANGELOG and README files in the various packages which we could also lint, as well as a SECURITY.md file at the top level. If you have the time to script linting them too, and they pass, I would add that.

Anything that doesn't pass we could add to the linter later. Ideally as we make changes to those readmes/other markdown files. (Granted the changelogs are automatically generated and may not need linting at all).

**Review by PeterSP — APPROVED** [review]

**Comment by mrsimpson** [comment]
@PeterSP @avatarneil I checked the existing markdown files and in fact there was not too much to be done (except figuring out that the `markdown-cli2` package is way better at ignoring `node_modules`).
The linter will now respect all `*.md` except the generated changelogs (one should never lint generated files, imho).
Feel free to merge if you feel comfortable, I consider this done now.

**Comment by mrsimpson** [comment]
When checking the change set, I noticed that

- when adding the `markdown-cli2` packages, a new package-lock was created. How do you usually proceed here: you you check it in with each PR?
- The pre-commit first wanted to create a new change set (for all packages) which I aborted since it's only readmes. How do you handle pre-commit and change sets?

**Comment by avatarneil** [comment]
@mrsimpson could you update this branch by pulling the latest `develop` down, doing a `git merge develop`, and then running `pnpm clean` (to resolve the pnpm-lock issues)? Also, you shouldn't have to worry about Jenkins being a bother anymore, we've switched over to Github Actions now :)

**Comment by mrsimpson** [comment]
@avatarneil sounds great!
Merged `develop` (I like you don't require rebasing, but instead squash, btw!), but GH says
> auto-merge was automatically disabled [1 minute ago](https://github.com/lacuna-tech/mds-core/pull/896#event-6136114466)
> Head branch was pushed to by a user without write access

**Pull request merged**